pkg/storaged/smart-details.jsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React, { useState } from "react";
+import * as timeformat from "timeformat.js";
+
+import { Card, CardBody, CardHeader, CardTitle, DescriptionList, DescriptionListDescription, DescriptionListGroup, DescriptionListTerm } from "@patternfly/react-core";
+import { Dropdown, DropdownItem, KebabToggle } from '@patternfly/react-core/dist/esm/deprecated/components/Dropdown/index.js';
+
+const _ = cockpit.gettext;
+
+const selftestStatusDescription = {
+    success: _("Successful"),
+    aborted: _("Aborted"),
+    interrupted: _("Interrupted"),
+    fatal: _("Did not complete"),
+    error_unknown: _("Failed (Unknown)"),
+    error_electrical: _("Failed (Electrical)"),
+    error_servo: _("Failed (Servo)"),
+    error_read: _("Failed (Read)"),
+    error_handling: _("Failed (Damaged)"),
+    inprogress: _("In progress"),
+};
+
+const SmartActions = ({ smartInfo }) => {
+    const [isKebabOpen, setKebabOpen] = useState(false);
+    const smartSelftestStatus = smartInfo.SmartSelftestStatus;
+
+    const runSmartTest = async (type) => {
+        await smartInfo.SmartSelftestStart(type, {});
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19103-20230717-115949-00eb709f-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)